broker/test/init

```diff
@@ -2,94 +2,68 @@
 
     testBroker="test/test_broker/*.js" #for global.setup.js 
     app=$(ls applications/**/test/*.js applications/**/test/**/*.js | grep -v "^applications/deployment_hooks")
     core="core/**/test/*.js"
     dataAccessLayer="data-access-layer/**/test/*.js"
     brokerOpts="test/test_broker/mocha_broker.opts"
-    
-    case $1 in 
-        "test")             _mocha $testBroker $app $core $dataAccessLayer --opts $brokerOpts;;
-        "test-ci")          babel-node ./node_modules/.bin/isparta cover --report cobertura _mocha -- $testBroker $app $core $dataAccessLayer --opts $brokerOpts;;
-        "test-coverall")    babel-node ./node_modules/.bin/isparta cover --report lcovonly _mocha -- $testBroker $app $core $dataAccessLayer --opts $brokerOpts;;
-        "test-cov")         babel-node ./node_modules/.bin/isparta cover _mocha -- $testBroker $app $core $dataAccessLayer --opts $brokerOpts;;
-    esac
-}
-
-test_deployment_hooks(){
-    rm -f ./logs/test.log
-
     testHooks="applications/deployment_hooks/test/*.js"
     hooksOpts="applications/deployment_hooks/test/mocha_deployment_hooks.opts"
-
+    
     case $1 in 
-        "test")             _mocha $testHooks --opts $hooksOpts;;
-        "test-ci")          babel-node ./node_modules/.bin/isparta cover --report cobertura _mocha  -- $testHooks --opts $hooksOpts;;
-        "test-coverall")    babel-node ./node_modules/.bin/isparta cover --report lcovonly _mocha -- $testHooks --opts $hooksOpts;;
-        "test-cov")         babel-node ./node_modules/.bin/isparta cover _mocha -- $testHooks --opts $hooksOpts;;
+        "test")             _mocha $testBroker $app $core $dataAccessLayer --opts $brokerOpts;
+                            _mocha $testHooks --opts $hooksOpts;;
+        
+        "test-ci")          babel-node ./node_modules/.bin/isparta cover --report cobertura _mocha -- $testBroker $app $core $dataAccessLayer --opts $brokerOpts || exit 1;
+                            mv ./coverage/cobertura-coverage.xml ./coverage/cobertura-coverage-broker.xml
+                            mv ./coverage/coverage.json ./coverage/coverage-broker.json
+                            babel-node ./node_modules/.bin/isparta cover --report cobertura _mocha  -- $testHooks --opts $hooksOpts || exit 1;;
+        
+        "test-coverall")    babel-node ./node_modules/.bin/isparta cover _mocha -- $testBroker $app $core $dataAccessLayer --opts $brokerOpts || exit 1;
+                            mkdir ./coverage/broker_report
+                            mv ./coverage/lcov.info ./coverage/lcov_broker.info
+                            mv -v ./coverage/coverage.json ./coverage/lcov-report/ ./coverage/broker_report
+                            babel-node ./node_modules/.bin/isparta cover _mocha -- $testHooks --opts $hooksOpts || exit 1;
+                            mkdir ./coverage/hooks_report
+                            mv -v ./coverage/coverage.json ./coverage/lcov-report/ ./coverage/hooks_report/ ;;
```

Comment:
Now, our developers can directly use `npm run test-coverall` instead of `npm run test-cov <broker/hooks>` to see the html reports too and it will send the information to coveralls. Two directories will be created inside `coverage` directory, `broker_report` and `hooks_report`. Inside these, we can see the html reports of respective suite.